src/gamestate/commands.cpp

```diff
@@ -430,7 +430,7 @@ void execute_begin_province_building_construction(sys::state& state, dcon::natio
 			si.set_naval_base_is_taken(true);
 	}
 
-	if(type == economy::province_building_type::railroad && source != state.world.province_get_nation_from_province_ownership(p)) {
+	if(type != economy::province_building_type::fort && type != economy::province_building_type::naval_base && source != state.world.province_get_nation_from_province_ownership(p)) {
```

Comment:
probably should stick with checking for RR here, unless TGC has ui for doing foreign investment with the new buildings??

Reply:
yes it does